Wed Mar 29 00:03:50 CEST 2006  kentwilliams
  * Insight: changeset 2006-03-28 22:03:50 by kentwilliams
  COMP: WrapITK requires numeric traits for itk::RGB_Pixel<unsigned
  short>
diff -rN -u old-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.cxx new-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.cxx
--- old-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.cxx	2006-04-05 17:07:45.947312361 +0200
+++ new-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.cxx	2006-03-30 01:50:31.000000000 +0200
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkNumericTraitsRGBPixel.cxx,v $
   Language:  C++
-  Date:      $Date: 2005/01/09 15:58:08 $
-  Version:   $Revision: 1.3 $
+  Date:      $Date: 2006/03/28 22:03:50 $
+  Version:   $Revision: 1.4 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -21,5 +21,7 @@
 
 const RGBPixel<unsigned char>  NumericTraits<RGBPixel<unsigned char> >::Zero = RGBPixel<unsigned char>( NumericTraits<unsigned char>::Zero );
 const RGBPixel<unsigned char>  NumericTraits<RGBPixel<unsigned char> >::One = RGBPixel<unsigned char>( NumericTraits<unsigned char>::One );
+const RGBPixel<unsigned short>  NumericTraits<RGBPixel<unsigned short> >::Zero = RGBPixel<unsigned short>( NumericTraits<unsigned short>::Zero );
+const RGBPixel<unsigned short>  NumericTraits<RGBPixel<unsigned short> >::One = RGBPixel<unsigned short>( NumericTraits<unsigned short>::One );
  
 } // end namespace itk
diff -rN -u old-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.h new-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.h
--- old-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.h	2006-04-05 17:07:45.946312671 +0200
+++ new-itk/Insight/Code/Common/itkNumericTraitsRGBPixel.h	2006-03-30 01:50:32.000000000 +0200
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkNumericTraitsRGBPixel.h,v $
   Language:  C++
-  Date:      $Date: 2005/08/08 19:18:22 $
-  Version:   $Revision: 1.8 $
+  Date:      $Date: 2006/03/28 22:03:51 $
+  Version:   $Revision: 1.9 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -42,6 +42,25 @@
   static ValueType ZeroValue() { return Zero; }
   
 };
+/** \class NumericTraits<RGBPixel<unsigned short> >
+ * \brief Define traits for type RGBPixel<unsigned char>.
+ * \ingroup DataRepresentation
+ */
+template <>
+class NumericTraits<RGBPixel<unsigned short> > {
+public:
+  typedef RGBPixel<unsigned char> ValueType;
+  typedef RGBPixel<unsigned char> PrintType;
+  typedef RGBPixel<unsigned char> AbsType;
+  typedef RGBPixel<unsigned short> AccumulateType;
+  typedef RGBPixel<double> RealType;
+  typedef double           ScalarRealType;
+  typedef RGBPixel<float>  FloatType;
+  static const RGBPixel<unsigned short> ITKCommon_EXPORT Zero;
+  static const RGBPixel<unsigned short> ITKCommon_EXPORT One;
+  static ValueType ZeroValue() { return Zero; }
+  
+};
 
 } // end namespace itk
 

